tests/cpp-tests/Classes/ClickAndMoveTest/ClickAndMoveTest.cpp

```diff
@@ -48,19 +48,23 @@ void ClickAndMoveTestCase::onTouchEnded(Touch* touch, Event  *event)
     auto location = touch->getLocation();
 
     auto s = getChildByTag(kTagSprite);
-    s->stopAllActions();
-    s->runAction( MoveTo::create(1, Vec2(location.x, location.y) ) );
-    float o = location.x - s->getPosition().x;
-    float a = location.y - s->getPosition().y;
-    float at = (float) CC_RADIANS_TO_DEGREES( atanf( o/a) );
     
-    if( a < 0 ) 
+    if (s->getPosition().x != location.x || s->getPosition().y != location.y)
     {
-        if(  o < 0 )
-            at = 180 + fabs(at);
-        else
-            at = 180 - fabs(at);    
+        s->stopAllActions();
+        s->runAction(MoveTo::create(1, Vec2(location.x, location.y)));
+        float o = location.x - s->getPosition().x;
+        float a = location.y - s->getPosition().y;
+        float at = (0.f == a) ? 90.f : (float)CC_RADIANS_TO_DEGREES(atanf(o / a));
```

Comment:
Why it is 90 if `y` is not changed? I think it should be 0, because it will run along x axis.